### PR TITLE
Added ARD Mediathek presence

### PR DESCRIPTION
**Pull request by Reichmann-M**
# Proof of work
[www.ardmediathek.de](https://www.ardmediathek.de/) - Mainpage of German Television Streaming portal
screen1 

[Das Erste Livestream](https://www.ardmediathek.de/live/Y3JpZDovL2Rhc2Vyc3RlLmRlL0xpdmVzdHJlYW0tRGFzRXJzdGU/?toolbarType=default) - German Television Livestream
screen2

[ARD Video-on-demand](https://www.ardmediathek.de/video/agatha-christie-partners-in-crime/folge-1-der-auftrag-s01-e01/one/Y3JpZDovL3dkci5kZS9CZWl0cmFnLWI1MzBjYzgyLTc1NjgtNDdjYi1hMjRjLWIyZTJmZGZlOTlmNQ/) - Normal show & movie streaming
screen3

### CODE REVIEW

File: websites/A/ARD Mediathek/presence.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  const presenceData: PresenceData = {
+    largeImageKey: "ard_mediathek"
+  };
+
+  if (document.location.href !== prevUrl) {
+    prevUrl = document.location.href;
+    elapsed = Math.floor(Date.now() / 1000);
+  }
+
+  const path = location.pathname.replace(/\/?$/, "/");
```
Comment (ririxidev) [inline websites/A/ARD Mediathek/presence.ts:26]:
```suggestion
  }, path = location.pathname.replace(/\/?$/, "/");


  if (document.location.href !== prevUrl) {
    prevUrl = document.location.href;
    elapsed = Math.floor(Date.now() / 1000);
  }
```

File: websites/A/ARD Mediathek/dist/metadata.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{
+    "$schema": "https://schemas.premid.app/metadata/1.3",
+    "service": "ARD Mediathek",
+    "author": {
+      "name": "TONi_C137#1757",
+      "id": "328213965299515393"
+    },
+    "category": "videos",
+    "logo": "https://upload.wikimedia.org/wikipedia/de/thumb/a/ad/ARD_Mediathek-Logo.png/600px-ARD_Mediathek-Logo.png",
+    "thumbnail": "https://www.daserste.de/unterhaltung/service-unterhaltung/ard-mediathek-das-erste-100~_v-standard644_d3b97e.jpg",
```
Comment (ririxidev) [inline websites/A/ARD Mediathek/dist/metadata.json:10]: use direct imgur links

File: websites/A/ARD Mediathek/presence.ts [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        parseInt(
+          document
+            .querySelector(".ardplayer-controlbar")
+            .getAttribute("aria-valuemax")
+        ) -
+        parseInt(
+          document
+            .querySelector(".ardplayer-controlbar")
+            .getAttribute("aria-valuenow")
+        );
```
Comment (ririxidev) [inline websites/A/ARD Mediathek/presence.ts:90]: presence.getTimestamps ?

**Review by rxri — CHANGES_REQUESTED** [review]

File: websites/A/ARD Mediathek/dist/metadata.json (hunk 2 above)
Comment (Reichmann-M) [reply websites/A/ARD Mediathek/dist/metadata.json:10]:
```suggestion
    "logo": "https://i.imgur.com/baUokNs.png",
    "thumbnail": "https://i.imgur.com/mix2mJC.jpg",
```

File: websites/A/ARD Mediathek/presence.ts (hunk 3 above)
Comment (Reichmann-M) [reply websites/A/ARD Mediathek/presence.ts:89]:
```suggestion
const video: HTMLVideoElement = document.querySelector('.ardplayer-mediacanvas'),
    timestamps = presence.getTimestamps(Math.floor(video.currentTime),Math.floor(video.duration));

presenceData.startTimestamp = timestamps[0];
presenceData.endTimestamp = timestamps[1];
```

**Comment by Reichmann-M** [comment]
Every change suggestion should be done by now ^^

File: websites/A/ARD Mediathek/presence.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+      const videoDateDIV = document.querySelector(".Line-epbftj-1.dgMIUj");
+      presenceData.state = `${
+        videoDateDIV.children[0].innerHTML
+      } from ${videoDateDIV.innerHTML.substring(
+        0,
+        videoDateDIV.innerHTML.indexOf("∙") - 1
+      )}`;
+
+      [presenceData.startTimestamp, presenceData.endTimestamp] = timestamps;
```
Comment (ririxidev) [inline websites/A/ARD Mediathek/presence.ts:81]:
```suggestion
      [, presenceData.endTimestamp] = timestamps;
```

**Review by rxri — CHANGES_REQUESTED** [review]
please run prettier over it :^)

**Review by rxri — APPROVED** [review]

**Review by Slowlife01 — APPROVED** [review]

**Pull request merged**